app/controllers/groups_controller.rb

```diff
@@ -73,7 +72,9 @@ def rename_group
       params[:assignment_id] = @assignment.id
 
       if Grouping.exists?(assessment_id: @assignment.id, group_id: groupexist_id)
-        flash_now(:error, I18n.t('groups.group_name_already_in_use'))
+        flash.now[:error] = I18n.t('groups.group_name_already_in_use')
+      elsif @existing_group.has_non_empty_submission?
```

Comment:
Overall, this check is not exactly what I'm looking for.

1. The check should centre on `@grouping`, not `@existing_group`. The original issue was that files that had originally been submitted to `@grouping` could no longer be accessed after the repository switched, and so the check I want you to make is for whether `@grouping` has any submitted files.
2. It's great research that you found the `Grouping#has_non_empty_submission?` method, but that's actually too narrow for our purposes. A `Submission` object is only created after an instructor "collects" submissions, typically after the assignment deadline. However, students submit files before this collection process, and the check I'd like you to make is for whether students have submitted -- i.e., _uploaded_ -- any files to `@grouping`.